src/sqlancer/clickhouse/gen/ClickHouseExpressionGenerator.java

```diff
@@ -365,12 +384,20 @@ public ClickHouseExpressionGenerator setTablesAndColumns(AbstractTables<ClickHou
         return this;
     }
 
-    @Override
-    public ClickHouseExpression generateBooleanExpression() {
-        List<ClickHouseColumnReference> columnRefs = columns.stream()
-                .map(c -> c.asColumnReference(c.getTable().getName())).collect(Collectors.toList());
-        return generateExpressionWithColumns(columnRefs, 5);
+ @Override
+public ClickHouseExpression generateBooleanExpression() {
+    List<ClickHouseColumnReference> columnRefs = columns.stream()
+            .map(c -> c.asColumnReference(c.getTable().getName()))
+            .filter(c -> c.getColumn().getType().getType() == ClickHouseDataType.UInt8) // Ensure only UInt8 columns
```

Comment:
This is meant not to select only Uint8 columns. Expression `c > 'abc'` is boolean, but can use any column.